tests/db_migrator_test.py

```diff
@@ -0,0 +1,178 @@
+import os
+import pytest
+import sys
+
+from swsssdk import SonicV2Connector
+from sonic_py_common import device_info
+
+from .mock_tables import dbconnector
+
+import config.main as config
+from utilities_common.db import Db
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+mock_db_path = os.path.join(test_path, "db_migrator_input")
+modules_path = os.path.dirname(test_path)
+scripts_path = os.path.join(modules_path, "scripts")
+sys.path.insert(0, test_path)
+sys.path.insert(0, modules_path)
+sys.path.insert(0, scripts_path)
+
+os.environ["PATH"] += os.pathsep + scripts_path
+
+def get_sonic_version_info_mlnx():
+    return {'asic_type': 'mellanox'}
+
+
+class TestMellanoxBufferMigrator(object):
+    @classmethod
+    def setup_class(cls):
+        cls.config_db_tables_to_verify = ['BUFFER_POOL', 'BUFFER_PROFILE', 'BUFFER_PG', 'DEFAULT_LOSSLESS_BUFFER_PARAMETER', 'LOSSLESS_TRAFFIC_PATTERN', 'VERSIONS', 'DEVICE_METADATA']
+        cls.appl_db_tables_to_verify = ['BUFFER_POOL_TABLE:*', 'BUFFER_PROFILE_TABLE:*', 'BUFFER_PG_TABLE:*', 'BUFFER_QUEUE:*', 'BUFFER_PORT_INGRESS_PROFILE_LIST:*', 'BUFFER_PORT_EGRESS_PROFILE_LIST:*']
+        cls.warm_reboot_from_version = 'version_1_0_5'
+        cls.warm_reboot_to_version = 'version_2_0_0'
```

Comment:
Why are we only covering warm reboot between these 2 versions? We need to include all possible to/from versions to avoid regression

Reply:
We don't need it because they are covered by cold reboot test. This results from the principle of the db_migrator.
Suppose the switch is warm-rebooted from version 1_0_2 and the latest version is 2_0_0. There are version 1_0_3, 1_0_4 and 1_0_5 in between.
The db_migrator will migrate the database in the following steps:
1. 1_0_2 to 1_0_3
2. 1_0_3 to 1_0_4
3. 1_0_4 to 1_0_5
4. 1_0_5 to 2_0_0

Only when migrated from 1_0_5 to 2_0_0 does db_migrator have dedicated handling for warm reboot. For any other steps, it isn't interested whether it's warm reboot or cold reboot and handles warm reboot and cold reboot in the exact same flow.
As we already have the test case for migrating from any version to 2_0_2 for cold reboot, we don't need to have them for warm reboot again.